module/observations/helper.js

```diff
@@ -61,77 +63,117 @@ module.exports = class ObservationsHelper {
+                    templateInformation.solution,
+                    organisationAndRootOrganisation
+                );
 
-                if (!solutionDocument) {
-                    throw new Error(messageConstants.apiResponses.SOLUTION_NOT_FOUND);
-                }
+                return resolve(_.pick(observationData, ["_id", "name", "description"]));
+
+            } catch (error) {
+                return reject(error);
+            }
+        })
+    }
+
+    /**
+     * Create observation.
+     * @method
+     * @name createObservation
+     * @param {Object} data - Observation creation data.
+     * @param {String} userId - Logged in user id.
+     * @param {Object} solution - Solution detail data.
+     * @param {Object} solution - Solution detail data.
+     * @param {String} organisationAndRootOrganisation - organisation and root organisation details. 
+     * @returns {Object} observation creation data.
+     */
+
+    static createObservation(data,userId,solution,organisationAndRootOrganisation) {
+        return new Promise(async (resolve, reject) => {
+            try {
 
                 if (data.entities) {
-                    let entitiesToAdd = await entitiesHelper.validateEntities(data.entities, solutionDocument.entityTypeId);
+                    let entitiesToAdd = 
+                    await entitiesHelper.validateEntities(data.entities, solution.entityTypeId);
                     data.entities = entitiesToAdd.entityIds;
                 }
 
-                let observationData = await database.models.observations.create(
+                let observationData = 
+                await database.models.observations.create(
                     _.merge(data, {
-                        "solutionId": solutionDocument._id,
-                        "solutionExternalId": solutionDocument.externalId,
-                        "frameworkId": solutionDocument.frameworkId,
-                        "frameworkExternalId": solutionDocument.frameworkExternalId,
-                        "entityTypeId": solutionDocument.entityTypeId,
-                        "entityType": solutionDocument.entityType,
-                        "author": userDetails.id,
-                        "updatedBy": userDetails.id,
-                        "createdBy": userDetails.id,
-                        "createdFor": createdFor,
-                        "rootOrganisations": rootOrganisations
+                        "solutionId": solution._id,
+                        "solutionExternalId": solution.externalId,
+                        "programId" : solution.programId,
+                        "programExternalId" : solution.programExternalId,
+                        "frameworkId": solution.frameworkId,
+                        "frameworkExternalId": solution.frameworkExternalId,
+                        "entityTypeId": solution.entityTypeId,
+                        "entityType": solution.entityType,
+                        "author": userId,
```

Comment:
@aman-tunerlabs - Why is this new field here? Isn't author just in a soltuion?

Reply:
@aks30  This was already there in existing observations/helper/create.

Reply:
@aman-tunerlabs - But is it used?

Reply:
@aks30  author is not there in observation model which means its not used

Reply:
@aman-tunerlabs - So is it being set when there is no such key in model?

Reply:
@aks30 Its not been set now . Its just an existing things which is been used when creating a new api.

Reply:
@aman-tunerlabs - So let's fix it old and new. Continuing on the existing way without it being needed doesn't make sense